### PR TITLE
Separate remote mirror target

### PR DESCRIPTION
**Pull request by bitjammer**

### CODE REVIEW

**Comment by bitjammer** [comment]
@swift-ci Please smoke test

File: utils/SwiftBuildSupport.py [hunk 1]
```diff
@@ -143,30 +143,35 @@ def get_preset_options(substitutions, preset_file_names, preset_name):
...
+            elif sdk == "TVOS_SIMULATOR":
+                tgts += StdlibDeploymentTarget.AppleTVSimulator.targets
+            elif sdk == "WATCHOS":
+                tgts += StdlibDeploymentTarget.AppleWatch.targets
+            elif sdk == "WATCHOS_SIMULATOR":
+                tgts += StdlibDeploymentTarget.AppleWatchSimulator.targets
+
+        build_script_opts.append("--stdlib-deployment-targets=" +
                                      " ".join([tgt.name for tgt in tgts]))
+    except IndexError:
```
Comment (gribozavr) [inline utils/SwiftBuildSupport.py]: This looks like a hack that can hide real issues, can we scope it down to the problematic statement?

File: stdlib/public/CMakeLists.txt [hunk 2]
```diff
@@ -31,6 +31,9 @@ if(SWIFT_BUILD_STDLIB)
   add_subdirectory(stubs)
   add_subdirectory(core)
   add_subdirectory(SwiftOnoneSupport)
+endif()
+
+if(SWIFT_BUILD_STDLIB OR SWIFT_BUILD_REMOTE_MIRROR)
```
Comment (gribozavr) [inline stdlib/public/CMakeLists.txt]: Please declare the new option with `option()`.

File: utils/SwiftBuildSupport.py (hunk 1 above)
Comment (bitjammer) [inline utils/SwiftBuildSupport.py]: Sure.

File: stdlib/public/CMakeLists.txt (hunk 2 above)
Comment (bitjammer) [inline stdlib/public/CMakeLists.txt]: Done, thanks.

**Comment by bitjammer** [comment]
@swift-ci Please smoke test

**Pull request merged**